neqo-http3/src/lib.rs

```diff
@@ -325,6 +332,43 @@ pub(crate) trait RecvMessageEvents: Debug {
     fn header_ready(&self, stream_id: u64, headers: Vec<Header>, interim: bool, fin: bool);
     fn data_readable(&self, stream_id: u64);
     fn reset(&self, stream_id: u64, error: AppError, local: bool);
+    fn web_transport_new_session(&self, stream_id: u64, headers: Vec<Header>);
+}
+
+pub trait WtRecvStream: RecvStream {
+    fn read_data(&mut self, conn: &mut Connection, buf: &mut [u8]) -> Res<(usize, bool)>;
+}
+
+pub trait WtSendStream: Debug {
+    fn send_data(&mut self, conn: &mut Connection, buf: &[u8]) -> Res<usize>;
+    fn close(&mut self, conn: &mut Connection) -> Res<()>;
+}
+
+pub trait WtEvents: Debug {
+    fn web_transport_session_negotiated(&self, stream_id: u64, success: bool);
+    fn web_transport_new_stream(&self, stream_id: u64);
+    fn web_transport_data_readable(&self, stream_id: u64);
+    fn web_transport_stream_reset(&self, stream_id: u64, error: AppError);
+    fn web_transport_data_writable(&self, stream_id: u64);
+    fn web_transport_stream_stop_sending(&self, stream_id: u64, error: AppError);
+    fn clone_box(&self) -> Box<dyn WtEvents>;
+}
+
+pub trait SendStream: Debug {
+    fn send(&mut self, conn: &mut Connection) -> Res<()>;
+    fn has_data_to_send(&self) -> bool;
+    fn stream_writable(&self);
+    fn done(&self) -> bool;
+    fn stop_sending(&mut self, error: AppError);
+    fn http_stream(&mut self) -> Option<&mut dyn HttpSendStream>;
```

Comment:
This (and the two after), could benefit from a default implementation here now that we have a few more places that this is implemented as:

```suggestion
    fn http_stream(&mut self) -> Option<&mut dyn HttpSendStream> { None }
```